pymess/functions.py

```diff
@@ -0,0 +1,55 @@
+from django.db import transaction
+
+
+def create_message(backend_fnc, recipient, content, related_objects, tag, template, **kwargs):
+    """
+    Create message which will be logged in the database.
+    :param backend_fnc: get backend function which will be evaluated based on recipient
+    :param recipient: email or phone number of the recipient
+    :param content: content of the message
+    :param related_objects: list of related objects that will be linked with the message using generic
+    relation
+    :param tag: string mark that will be saved with the message
+    :param template: template object from which content of the message was created
+    :param kwargs: extra attributes that will be saved with the message
+    """
+    return backend_fnc(recipient=recipient).create_message(recipient, content, related_objects, tag, template, **kwargs)
```

Comment:
why this shoud do backend inside

Reply:
In general this should work as expected without finding the correct backend assuming for each backend types is the same implementation. But to make it 100% correct it should call  `create_message` for the right backend IMO - `create_message` is a custom method of concrete backend and can be overridden.

Reply:
with controller separation this is not needed.

Reply:
refactored